### ISSUE TITLE
Japanese Region/Language emulation not working

### ISSUE THREAD

**Issue by MegaPramuk** [chaoskagami/corbenik#53]
I don't know if region/language emulation isn't working at all or if it's just for Japanese games but the two games that I've tested (only two Japanese games with DLC I possess) tell me that not every DLC could be read. But none of the DLCs are showing up.
The games in question are 000400000102F00 and 000400000190E00 (too lazy for Shift-JIS, it's Taiko no Tatsujin 2 and 3), which I've tried running on a European o3DS and European o3DS XL.
The DLCs' are dumped from my Japanese n3DS XL where they work fine, even if I delete both ticket and DLC from it and install the cia files.
Unfortunately, I don't have any European DLC I could test on the n3DS XL for the region emulation.
Also, I've tried to change the JA in both locale files to JP since I thought that it could have been because of 3dsdb.com's labeling of Japanese as ja instead of jp. But that didn't help either.
Not sure if it will help but [here](http://pastebin.com/SmeQvL1y)'s a verbose output of the boot.log, perhaps some patch is interfering with the emulation (which I doubt but hey, some even stranger bugs happened).

**Comment by chaoskagami** [comment]
boot.log will not aid me in debugging this issue. Language patching is performed by loader, not the initial boot process (and thus, that's the wrong log.)

Please upload loader.log.

Also, changing it to JP will cause it to go unpatched - you want JA, trust me. This is the list of constants: https://github.com/chaoskagami/corbenik/blob/master/external/loader/source/patcher.c#L141

**Comment by MegaPramuk** [comment]
I don't have a loader.log in /corbenik/var/log or anywhere else for the matter on neither of the devices as it seems.
Despite the option to log being on (hence the boot.log). It only generates a boot.log in said directory.

**Comment by chaoskagami** [comment]
That's literally impossible.

boot.log says the loader module was injected, and if you have verbose+logging on, loader *should* be logging to /var/log/loader.log.

If it isn't, the only possiblility is an improper upgrade where loader.cxi and arm9loaderhax.bin are mismatched, or some unknown bug preventing logging (and I can't repro that)

**Comment by MegaPramuk** [comment]
I've just rebuilt corbenik from the latest commit, deleted the whole /corbenik directory and arm9loaderhax.bin, because when you mentioned the incompatibilty I thought that I didn't update the arm9loaderhax.bin when updating corbenik.
So I put the corbenik folder and the arm9loaderhax.bin on the root.
And things didn't change. Region emulation is still not working and there's no loader.log.

**Comment by chaoskagami** [comment]
Can you zip up the contents of out/ from your build tree and upload that somewhere? I suspect since loader isn't logging there's something more fundamentally wrong here.

What OS are you building on?

EDIT: And if it's Windows, are you using native devkitPro+msys2, cygwin, or WSL to build it?

EDIT EDIT: I won't be able to get back here until I sleep some, but I'll try my best to get back to you in a timely manner when I wake up tomorrow. :/

**Comment by MegaPramuk** [comment]
Sure, here you go.
-removed-
This has been compiled on my Gentoo machine.
And don't worry, this isn't a really high priority issue (for me at least) any way since I can just play the DLC songs on my n3DS. Just get some rest.

Edit: I've changed the link because I didn't include the o3DS FIRMs before, not that it matters much that they're now there.

**Comment by chaoskagami** [comment]
Honestly, I'm more concerned overall that loader isn't logging on your 3DS right now. It's impossible to debug this without the loader.log, which for some reason...doesn't exist. I can't identify anything that would result in not logging aside from failing to read the actual configuration file (and that would disable everything aside from regular patches - which might explain why it isn't working)

Can you upload the contents of /corbenik/etc on your 3DS as it is on the console right now?

Nothing immediate jumps out at me from your binaries, which is good. Also; you did not need to include FIRMs. In fact, I would have preferred you didn't since now I have to remove that link for containing copyrighted content. :(

**Comment by MegaPramuk** [comment]
That was actually the reason why I didn't include them at first, but then I thought that it couldn't be a copyright infringement since they were still in their natural state and they're accessible without any kind of protection. Sorry!
https://jii.moe/SJpGybM8e.zip here's the link to /corbenik/etc/.
I've excluded the current-nand-cid, even if there's nothing stored and if the name's about what it implies then it wouldn't help you much either way. If it does however, just tell me.

**Comment by chaoskagami** [comment]
Okay, random question; did you remove the CID fragment from the end of the config file's name? Normally, they're named as `config-<NAND CID[0]>`. If the file is named literally `config-` on device, that points to the NAND CID reading code.

Reading through it briefly, I think there's an edge case when the CID begins with a `0`, which causes the configuration name to end up zero-terminated early in the boot process but not in loader. Can you dump current-nand-cid, please? Or at very least, is the first byte between 0x00 and 0x0F?

Side note: the CID handling never utilizes anything past the first four bytes and the CID totals 16 bytes. Since the file only stores the first four, one can't know the full CID of your console NAND with that file.

**Comment by chaoskagami** [comment]
Can you try building from the latest commit and tell me whether this resolves the logging issue (and potentially langemu, since I doubt it's even applied)

**Comment by MegaPramuk** [comment]
No, I didn't remove anything from the file. It's really called config-.
```
$ ls -l 3ds/etc/*
-rw-r--r-- 1 notebook notebook 1049608 Jan 10 07:24 config-
-rw-r--r-- 1 notebook notebook             4 Jan 10 07:24 current-nand-cid
```
And just fyi, the content of current-nand-cid:
```
$ cat current-nand-cid | xxd
00000000: cd10 cd0c                                ....
```
Anyway, I've compiled it from the latest commit and now everything works as intended. loader gets patched correctly and can show me the Taiko DLCs, the loader.log is finally in /var/log/ and the config- has a proper name now (which did start with a 0 btw).
At least on the o3DS, I can't test it on the o3DS XL right now since I've borrowed it to a friend yesterday.
Closing this issue now until I can test it out on the o3DS XL, but I believe that it won't be necessary.
Thank you very much!
